modules/apps/export-import/export-import-service/src/main/java/com/liferay/exportimport/internal/background/task/LayoutSetPrototypeImportBackgroundTaskExecutor.java

```diff
@@ -59,9 +60,8 @@ public LayoutSetPrototypeImportBackgroundTaskExecutor() {
 		setBackgroundTaskStatusMessageTranslator(
 			new LayoutExportImportBackgroundTaskStatusMessageTranslator());
 
-		// Isolation level guarantees this will be serial in a group
-
-		setIsolationLevel(BackgroundTaskConstants.ISOLATION_LEVEL_GROUP);
+		setIsolationLevel(
+			PropsValues.LAYOUT_SET_PROTOTYPE_IMPORT_TASK_ISOLATION);
```

Comment:
@vendeltoreki This executor is also in module, could you make this one an osgi configuration, too?

Reply:
Hi @dantewang ,

I tried to convert this property as well to a configuration value. However, since the ConfigurationProviderUtil is somehow not fully available when [BackgroundTaskExecutorConfigurator](https://github.com/liferay/liferay-portal/blob/master/modules/apps/export-import/export-import-service/src/main/java/com/liferay/exportimport/internal/background/task/BackgroundTaskExecutorConfigurator.java) is initialized during portal startup.

I get this error and the portal doesn't start up:
```
2022-10-03 14:41:37.542 ERROR [Start Level: Equinox Container: 93ff58aa-8c48-472a-9209-b68c38d9e92e][ServiceProxyFactory:278] Service "com.liferay.portal.kernel.module.configuration.ConfigurationProvider" is unavailable in 60000 milliseconds while setting field "_configurationProvider" for class "com.liferay.portal.kernel.module.configuration.ConfigurationProviderUtil", will retry...
2022-10-03 14:41:38.051 INFO  [Start Level: Equinox Container: 93ff58aa-8c48-472a-9209-b68c38d9e92e][ServiceProxyFactory:68] Running "Declarative Service Soft Circular Dependency Checker" check result: No issues were found.
2022-10-03 14:41:38.377 WARN  [Start Level: Equinox Container: 93ff58aa-8c48-472a-9209-b68c38d9e92e][ServiceProxyFactory:74] Running "Declarative Service Unsatisfied Component Checker" check result: _Bundle {id: 25, name: com.liferay.portal.lpkg.deployer.impl, version: 4.0.17}_
```
I was experimenting with system-level configuration as well. Unfortunately, it did not work either: https://github.com/vendeltoreki/liferay-portal/compare/master...vendeltoreki:LPS-150565-3-fix
I got the same error here as well.

Do you have any idea how this can be solved?

Reply:
@vendeltoreki Could you reference `ConfigurationProvider` in `LayoutSetPrototypeImportBackgroundTaskExecutor`? Basically the `Util` classes which wraps OSGi services should not be used in other OSGi services.

Reply:
Yes, I tried referencing it in [this experiment](https://github.com/vendeltoreki/liferay-portal/compare/master...vendeltoreki:LPS-150565-3-fix#diff-874dcc9824195ce33c252b295a64251016ace0752149783aa5672ca77119510fR115), but in this case, another reference (`SettingsLocatorHelper`) could not be resolved.
I tried it from [BackgroundTaskExecutorConfigurator](https://github.com/liferay/liferay-portal/blob/master/modules/apps/export-import/export-import-service/src/main/java/com/liferay/exportimport/internal/background/task/BackgroundTaskExecutorConfigurator.java) (it's still in the same `export-import-service` module), since `LayoutSetPrototypeImportBackgroundTaskExecutor` is not an OSGi component and is directly instantiated from here.
```
2022-10-06 10:38:47.211 ERROR [main][ServiceProxyFactory:278] Service "com.liferay.portal.kernel.settings.SettingsLocatorHelper" is unavailable in 60000 milliseconds while setting field "_settingsLocatorHelper" for class "com.liferay.portal.kernel.settings.SettingsLocatorHelperUtil", will retry...
2022-10-06 10:38:48.017 INFO  [main][ServiceProxyFactory:68] Running "Declarative Service Soft Circular Dependency Checker" check result: No issues were found.
2022-10-06 10:38:49.946 WARN  [main][ServiceProxyFactory:74] Running "Declarative Service Unsatisfied Component Checker" check result: ...
```

Reply:
@vendeltoreki Could you send me a PR with what you changed? I'll take a look.

Reply:
Thanks, @dantewang . I'm cleaning it up a little bit and sending it to you.